packages/tab/index.js

```diff
@@ -0,0 +1,192 @@
+    }
+  }
+
+  componentWillUnmount() {
+    this.foundation_.destroy();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.active !== prevProps.active) {
+      if (this.props.active) {
+        this.foundation_.activate(this.props.previousActiveClientRect);
+      } else {
+        this.foundation_.deactivate();
+      }
+    }
+  }
+
+  get classes() {
+    const {classList} = this.state;
+    const {className, minWidth, stacked} = this.props;
+    return classnames('mdc-tab', Array.from(classList), className, {
+      'mdc-tab--min-width': minWidth,
+      'mdc-tab--stacked': stacked,
+    });
+  }
+
+  get adapter() {
+    return {
+      addClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className) => this.classes.split(' ').includes(className),
+      setAttr: (attr, value) => this.setState({[attr]: value}),
+
+      registerEventHandler: () => this.allowTransitionEnd_ = true,
+      deregisterEventHandler: () => this.allowTransitionEnd_ = false,
+      getOffsetLeft: () => this.tabElement_.current && this.tabElement_.current.offsetLeft,
+      getOffsetWidth: () => this.tabElement_.current && this.tabElement_.current.offsetWidth,
+      getContentOffsetLeft: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetLeft,
+      getContentOffsetWidth: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetWidth,
+      focus: () => this.tabElement_.current && this.tabElement_.current.focus(),
+
+      // activateIndicator, deactivateIndicator, and computeIndicatorClientRect
+      // are not needed in the adapter. TabIndicator keeps calls foundation.active
+      // and foundation.deactivate when this.props.activate updates.
+      // computeIndicatorClientRect seems redundant in mdc-tab
+    };
+  }
+
+  computeIndicatorClientRect = () => {
+    if (!this.tabIndicatorElement_.current) return;
+    return this.tabIndicatorElement_.current.computeContentClientRect();
+  }
+
+  computeDimensions = () => {
+    return this.foundation_.computeDimensions();
+  }
+
+  handleTransitionEnd = (evt) => {
+    this.props.onTransitionEnd(evt);
+
+    if (!this.allowTransitionEnd_) return;
+
+    this.foundation_.handleTransitionEnd_(evt);
+  }
+
+  render() {
+    const {
+      /* eslint-disable */
+      active,
+      previousActiveClientRect,
+      className,
+      fadeIndicator,
+      indicator,
+      minWidth,
+      onTransitionEnd,
+      stacked,
+      /* eslint-enable */
+      children,
+      ...otherProps
+    } = this.props;
+
+    const {
+      tabIndex,
+      ['aria-selected']: ariaSelected,
+    } = this.state;
+
+    return (
+      <button
```

Comment:
the tab is a button??

Reply:
ya it is documented in MCW as a `<button>`. It makes sense to me, since the user's main action is to click it.